src/blockchain/rpc.rs

```diff
@@ -710,6 +786,38 @@ where
     Ok(())
 }
 
+fn pagenated_import<'a, S>(
+    client: &Client,
+    use_desc: bool,
+    start_epoch: u64,
+    poll_rate_sec: u64,
+    page_size: usize,
+    scripts_iter: S,
+    progress: &dyn Progress,
+) -> Result<(), Error>
+where
+    S: Iterator<Item = &'a Script> + Clone,
+{
+    (0_usize..)
+        .map(|page_index| {
+            scripts_iter
+                .clone()
```

Comment:
Does this clone all the elements in the iterator? If that's the case, maybe we should replace these few lines with a custom iterator that just consumes `scripts_iter` once and returns the item in batches.

Reply:
The iterator for `Vec` uses a slice (which is a reference). Also the elements here are references too, so I think this is okay.